### PR TITLE
Marathon Lib

### PR DESCRIPTION
**Pull request by Henrod**
This PR is for a lib for Marathon. The only implemented routes are CreateJob and ListJobs, although it contains the base to add all the API routes.

### CODE REVIEW

**Comment by Henrod** [comment]
Also, the Gopkg.lock changed when I ran dep ensure.

File: lib/lib.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+// using template and payload.
+func (m *Marathon) CreateJob(
+	ctx context.Context,
+	template string,
+	payload *CreateJobPayload,
+) (*Job, error) {
+	route := m.buildCreateJobURL(template)
+	response := &Job{}
+
+	err := m.sendTo(ctx, "POST", route, payload, response)
```
Comment (capella) [inline lib/lib.go]: Why not `http.MethodPost`?

File: lib/lib_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+		return string(bytes)
+	}
+
+	Describe("Creating new job", func() {
+		It("should create job", func() {
+			url := fmt.Sprintf("http://marathon/apps/%s/jobs", appID)
+			httpmock.RegisterResponder(
+				"POST", url,
+				httpmock.NewStringResponder(200, toString(map[string]interface{}{
```
Comment (capella) [inline lib/lib_test.go]: The response of a list of jobs will be `[]model.Job{}`. I think using the real response model with `httpmock.NewJsonResponder` will be better than using a `map[string]interface{}`.

File: lib/lib_test.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+			Expect(job).To(BeNil())
+		})
+	})
+
+	Describe("Listing jobs", func() {
+		It("should create job", func() {
+			url := fmt.Sprintf("http://marathon/apps/%s/jobs", appID)
+			httpmock.RegisterResponder(
+				"GET", url,
+				httpmock.NewStringResponder(200, toString([]interface{}{
```
Comment (capella) [inline lib/lib_test.go]: Same here.

**Comment by felipejfc** [comment]
For creating this kind of client lib I suggest using [go-swagger](https://github.com/go-swagger/go-swagger) because of code generation...
@codingllama can probably give more information about it if you like

**Comment by capella** [comment]
I think using `go-swagger` is a good idea, however, in my opinion, it is only useful when you have the server and the clients aligned. That is, both need to use the same specification (OpenAPI Specification). In this project, almost all the inputs and outputs for the servers are defined in models or are a list of models.

To simplify the things, in this project we can use the same the models, all the servers responses and inputs are mapped to models. Using models in the client will be a good solution because all the communication part will be defined in one place. I know that this is not the best practice, I prefer using protos with grpc-gateway, however, I think we don't have time do make a change like this in the project.

I think that Herod is in a hurry to get the communication with the API and the solution h is proposing is the best one. This project needs to be refractory and this change can be included when we make that.

**Comment by Henrod** [comment]
I agree with @capella

**Comment by capella** [comment]
LGTM

**Pull request merged**